util/imagetools/create.go

```diff
@@ -138,12 +138,25 @@ func (r *Resolver) Combine(ctx context.Context, srcs []*Source) ([]byte, ocispec
 		mt = ocispec.MediaTypeImageIndex
 	}
 
+	// annotations are only allowed on OCI indexes
+	annotations := make(map[string]string)
+	if mt == ocispec.MediaTypeImageIndex {
+		for k, v := range ann {
+			if !strings.HasPrefix(k, "index:") {
+				return nil, ocispec.Descriptor{}, errors.Errorf("invalid annotation key %q, expected it to have 'index:' prefix", k)
+			}
```

Comment:
Suggestion: let's change this to a switch statement, checking for all the variants: `index` is supported, `index-descriptor` doesn't make sense in this context (so we should return a suitable error message), `manifest-descriptor` and `manifest` should error as not currently supported, all others should error with an "unknown"-like error.